spring-kafka/src/main/java/org/springframework/kafka/core/KStreamBuilderFactoryBean.java

```diff
@@ -116,8 +130,12 @@ public synchronized void start() {
 		if (!this.running) {
 			try {
 				this.kafkaStreams = new KafkaStreams(getObject(), this.streamsConfig, this.clientSupplier);
-				this.kafkaStreams.setStateListener(this.stateListener);
-				this.kafkaStreams.setUncaughtExceptionHandler(this.exceptionHandler);
+				if (this.stateListener != null) {
```

Comment:
Yes, I understand; I checked that logic before merging; however, I thought about it overnight and I am concerned about...

    fb.getKafkaStreams().setStateListener(myStateListener);
    fb.start();

I agree the setters are useful (for XML Spring config perhaps), but I think the null check is needed for the above reason.